src/renderer/store.ts

```diff
@@ -1,19 +1,284 @@
-import { createStore, combineReducers } from 'redux';
-import * as LocalRenderer from '@getflywheel/local/renderer';
-import { IPC_EVENTS } from '../constants';
-import { Preferences } from '../types';
-import { preferencesReducer } from './reducers'
-
-
-const makeStore = async () => {
-	const preferences: Preferences = await LocalRenderer.ipcAsync(
-		IPC_EVENTS.READ_PREFERENCES_FROM_DISK
-	);
-
-	return createStore(
-		combineReducers({ preferences: preferencesReducer }),
-		{ preferences },
-	);
+import { useSelector, TypedUseSelectorHook } from 'react-redux';
+import {
+	configureStore,
+	createSlice,
+	createSelector,
+	PayloadAction,
+} from '@reduxjs/toolkit';
+
+import {
+	Preferences,
+	CachedImageDataBySiteID,
+	SiteImageData,
+	OptimizerStatus,
+	ImageData,
+	FileStatus,
+} from '../types';
+
+interface SiteActionPayload {
+	siteID: string;
+	siteImageData: SiteImageData;
+	error: Error;
+}
+
+function mergeSiteState(state, payload: Partial<SiteActionPayload>, newState?: Partial<SiteImageData>) {
+	const { siteID, siteImageData } = payload;
+
+	state[siteID] = {
+		...state[siteID],
+		...siteImageData,
+		...newState,
+	};
+
+	return state;
+}
```

Comment:
Super nice convenience method for something that's definitely a bit more complex where the abstraction keeps the implementation much cleaner!

Reply:
🙌